LoginApplication/app/src/main/java/com/example/webonise/loginapplication/SignUpActivity.java

```diff
@@ -0,0 +1,281 @@
+        spinner.setAdapter(dataAdapter);
+    }
+
+    @Override
+    public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
+        String item = parent.getItemAtPosition(position).toString();
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView<?> parent) {
+    }
+
+    @Override
+    public void onClick(View v) {
+
+        switch (v.getId()) {
+            case R.id.btnLogin:
+                onLoginButtonClicked();
+                break;
+            case R.id.btnCancel:
+                System.exit(0);
+                break;
+            default:
+                break;
+        }
+    }
+
+    private void onLoginButtonClicked() {
+
+        firstname = editFirstName.getText().toString();
+        lastname = editLastName.getText().toString();
+        contactString = editContact.getText().toString();
+        email = editEmail.getText().toString();
+        password = editPassword.getText().toString();
+        confirmPassword = editConfirmPassword.getText().toString();
+        address = editAddress.getText().toString();
+        securityQuestion = spinner.getSelectedItem().toString();
+        securityAnswer = editAnswer.getText().toString();
+
+        contactdouble = Double.parseDouble(contactString);
+        contact = (int) contactdouble;
+
+        if (!valiDateProfile()) {
+            Toast toast = Toast.makeText(getApplicationContext(), "Enter Data", Toast.LENGTH_SHORT);
+        } else {
+            saveToDatabase(firstname, lastname, contact, email, password, confirmPassword, address, securityQuestion, securityAnswer);
+            Intent intent = new Intent(this, ActivityMenu.class);
+            startActivity(intent);
+        }
+    }
+
+    private void saveToDatabase(final String fname, final String lname, final int contact, final String email, final String password, final String confirmPassword, final String address, final String securityQuestion, final String securityAnswer) {
+        realm.executeTransaction(new Realm.Transaction() {
+            @Override
+            public void execute(Realm bgRealm) {
+                UserProfile user = realm.createObject(UserProfile.class);
+                user.setFname(fname);
+                user.setLname(lname);
+                user.setContact(contact);
+                user.setEmail(email);
+                user.setPassword(password);
+                user.setConfirmPassword(confirmPassword);
+                user.setAddress(address);
+                user.setSecurityQuestion(securityQuestion);
+                user.setSecurityAnswer(securityAnswer);
+            }
+        });
+        Log.v("message", "Success on data storing");
+    }
+
+    private boolean valiDateProfile() {
+        String aplhaPattern = "([a-zA-Z]{3,30}\\s*)+";
+        String emailPattern = "[a-zA-Z0-9._-]+@[a-z]+\\.+[a-z]+";
+        if (firstname.trim().isEmpty()) {
+            input_layout_fname.setError("Enter First Name");
```

Comment:
Shouldn't use hardcoded string values. Take them out in strings.xml and use reference